add UI testing workflow with Maestro for APK

rfaturriza opened the pull request:
This pull request adds a new job to the Flutter CI workflow to automate UI testing using Maestro. The new `ui-tests` job runs after the build step, downloads the generated APK, and executes UI tests with reporting enabled.

**CI/CD workflow enhancements:**

* Added a `ui-tests` job to `.github/workflows/flutter_ci.yml` that runs Maestro UI tests on the built APK, including steps to check out the repository, download the APK artifact, and generate a test report.

seribasa-developer commented:
@rfaturriza 
- can we execute tests for several emulator-api, starting from 31 to the latest?
- can you create a sample, e.g. a login flow and show where the video is generated?
- can we export [test-reporter](https://github.com/marketplace/actions/test-reporter) action to show junit report on github?

rfaturriza commented:
> @rfaturriza
> 
> * can we execute tests for several emulator-api, starting from 31 to the latest?
> * can you create a sample, e.g. a login flow and show where the video is generated?
> * can we export [test-reporter](https://github.com/marketplace/actions/test-reporter) action to show junit report on github?

Review latest commit
- I added metrix for test emulator-api 31 -35 
- i added onboarding-flow for the sample test
- I added test report for each emulator-api and also add for merged report for all emulator-api

note: It seems for test report we need to give permission https://anayooleru.medium.com/resource-not-accessible-by-integration-on-github-actions-2024-solution-a8d1bd5d847f
